mmv1/third_party/terraform/tests/resource_compute_region_ssl_policy_test.go.erb

```diff
@@ -0,0 +1,232 @@
+	t.Parallel()
+
+	var sslPolicy compute.SslPolicy
+	sslPolicyName := fmt.Sprintf("test-ssl-policy-%s", randString(t, 10))
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeSslPolicyDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccComputeRegionSslUpdate3(sslPolicyName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeRegionSslPolicyExists(
+						t, "google_compute_region_ssl_policy.update", &sslPolicy),
+					resource.TestCheckResourceAttr(
+						"google_compute_region_ssl_policy.update", "profile", "CUSTOM"),
+					resource.TestCheckResourceAttr(
+						"google_compute_region_ssl_policy.update", "min_tls_version", "TLS_1_1"),
+				),
+			},
+			{
+				ResourceName:      "google_compute_region_ssl_policy.update",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccComputeRegionSslUpdate1(sslPolicyName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeRegionSslPolicyExists(
+						t, "google_compute_region_ssl_policy.update", &sslPolicy),
+					resource.TestCheckResourceAttr(
+						"google_compute_region_ssl_policy.update", "profile", "MODERN"),
+					resource.TestCheckResourceAttr(
+						"google_compute_region_ssl_policy.update", "min_tls_version", "TLS_1_0"),
+				),
+			},
+			{
+				ResourceName:      "google_compute_region_ssl_policy.update",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckComputeRegionSslPolicyExists(t *testing.T, n string, sslPolicy *compute.SslPolicy) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+
+		config := googleProviderConfig(t)
+
+		project, err := getTestProject(rs.Primary, config)
+		if err != nil {
+			return err
+		}
+
+		name := rs.Primary.Attributes["name"]
+
+		found, err := config.NewComputeClient(config.userAgent).RegionSslPolicies.Get(
+			project, "us-central1" , name).Do()
+		if err != nil {
+			return fmt.Errorf("Error Reading SSL Policy %s: %s", name, err)
+		}
+
+		if found.Name != name {
+			return fmt.Errorf("SSL Policy not found")
+		}
+
+		*sslPolicy = *found
+
+		return nil
+	}
+}
+
+func testAccComputeRegionSslUpdate1(resourceName string) string {
+	return fmt.Sprintf(`
+resource "google_compute_region_ssl_policy" "update" {
+  name            = "%s"
+  description     = "Generated by TF provider acceptance test"
+  min_tls_version = "TLS_1_0"
+  profile         = "MODERN"
+  region          = "us-central1"
+}
+`, resourceName)
+}
+
+func testAccComputeRegionSslUpdate2(resourceName string) string {
+	return fmt.Sprintf(`
+resource "google_compute_region_ssl_policy" "update" {
```

Comment:
nit: if you wanted, you could have one update function here and share with the above function, then just pass in the `profile` and `min_tls_version` values.

Reply:
I think the multiple update functions allow easy extension into different combinations of fields.